### PR TITLE
HTVDO-2699: adding clipping lambda infra

### PR DESCRIPTION
**Pull request by JuanFontes**
### Jira URL
https://thetower.atlassian.net/browse/HTVDO-2699

### Terraform plan on dev

```
Terraform will perform the following actions:

  # aws_lambda_permission.allow_result_bucket will be created
  + resource "aws_lambda_permission" "allow_result_bucket" {
      + action        = "lambda:InvokeFunction"
      + function_name = (known after apply)
      + id            = (known after apply)
      + principal     = "s3.amazonaws.com"
      + source_arn    = "arn:aws:s3:::enps-bucket"
      + statement_id  = "AllowExecutionFromS3Bucket"
    }

  # aws_s3_bucket_notification.s3_object_notification will be created
  + resource "aws_s3_bucket_notification" "s3_object_notification" {
      + bucket = "enps-bucket"
      + id     = (known after apply)

      + lambda_function {
          + events              = [
              + "s3:ObjectCreated:*",
            ]
          + filter_suffix       = ".xml"
          + id                  = (known after apply)
          + lambda_function_arn = (known after apply)
        }
    }

  # module.autoclipping_lambda.aws_iam_policy.cloudwatch_policy[0] will be created
  + resource "aws_iam_policy" "cloudwatch_policy" {
      + arn         = (known after apply)
      + description = "video-auto-clipping dev cloudwatch policy"
      + id          = (known after apply)
      + name        = "video-auto-clipping_dev_cloudwatch_policy"
      + path        = "/"
      + policy      = jsonencode(
            {
              + Statement = [
                  + {
                      + Action   = [
                          + "logs:CreateLogGroup",
                          + "logs:CreateLogStream",
                          + "logs:PutLogEvents",
                          + "logs:DescribeLogStreams",
                        ]
                      + Effect   = "Allow"
                      + Resource = [
                          + "arn:aws:logs:*:*:*",
                        ]
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
    }

  # module.autoclipping_lambda.aws_iam_policy_attachment.cloudwatch_policy_attachment[0] will be created
  + resource "aws_iam_policy_attachment" "cloudwatch_policy_attachment" {
      + id         = (known after apply)
      + name       = "video-auto-clipping_dev_cloudwatch_policy_attachment"
      + policy_arn = (known after apply)
      + roles      = [
          + "video-auto-clipping_dev_lambda_role",
        ]
    }

  # module.autoclipping_lambda.aws_iam_role.lambda_role[0] will be created
  + resource "aws_iam_role" "lambda_role" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = "lambda.amazonaws.com"
                        }
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + max_session_duration  = 3600
      + name                  = "video-auto-clipping_dev_lambda_role"
      + path                  = "/"
      + unique_id             = (known after apply)
    }

  # module.autoclipping_lambda.aws_iam_role_policy_attachment.vpc_policy_attachment[0] will be created
  + resource "aws_iam_role_policy_attachment" "vpc_policy_attachment" {
      + id         = (known after apply)
      + policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole"
      + role       = "video-auto-clipping_dev_lambda_role"
    }

  # module.autoclipping_lambda.aws_lambda_function.lambda_function[0] will be created
  + resource "aws_lambda_function" "lambda_function" {
      + arn                            = (known after apply)
      + filename                       = ".terraform/modules/autoclipping_lambda/lambda-deploy/build/app.zip"
      + function_name                  = "dev-video-auto-clipping"
      + handler                        = "main.autocliping_lambda"
      + id                             = (known after apply)
      + invoke_arn                     = (known after apply)
      + last_modified                  = (known after apply)
      + layers                         = []
      + memory_size                    = 128
      + publish                        = false
      + qualified_arn                  = (known after apply)
      + reserved_concurrent_executions = -1
      + role                           = (known after apply)
      + runtime                        = "python3.8"
      + source_code_hash               = "TQzYDZJJUsTA//cgLoyCAmQYcmDmVQF97ytsi1gx8Qg="
      + source_code_size               = (known after apply)
      + tags                           = {
          + "Application"  = "Lambda Function"
          + "BusinessUnit" = "HTVDigital"
          + "CostCenter"   = "40000014"
          + "Deprecate"    = "N"
          + "Environment"  = "Dev"
          + "Monitoring"   = "CloudWatch"
          + "Product"      = "video-auto-clipping"
          + "SupportTeam"  = "HTV_DevOps2@hearst.com"
          + "TechContact"  = "HTV DevOps"
        }
      + timeout                        = 3
      + version                        = (known after apply)

      + environment {
          + variables = {
              + "ENVIRONMENT"     = "dev"
              + "SERVED_STATIONS" = "KHTV"
              + "VAULT_ADDRESS"   = "https://vault.htvapps.net"
              + "VAULT_PASSWORD"  = "xx"
              + "VAULT_USERNAME"  = "xx"
            }
        }

      + tracing_config {
          + mode = (known after apply)
        }

      + vpc_config {
          + vpc_id = (known after apply)
        }
    }

Plan: 7 to add, 0 to change, 0 to destroy.
```

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import json
+
+def autocliping_lambda(event, context):
+    return {
```
Comment (mloudon) [inline src/main.py:4]: no curly brackets or parameterized return needed here - should perhaps just be 
```
def autocliping_lambda(event, context):
   print("hello world")
```

Comment (JuanFontes) [reply src/main.py:4]: oops, thanks @mloudon will fix!

File: deploy/terraform/lambda.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
+module "autoclipping_lambda" {
+  source      = "git::git@github.com:hearsttv/htv-terraform-modules//lambda-deploy?ref=LAMBDA-2.3.0"
```
Comment (vivekthakker) [inline deploy/terraform/lambda.tf:3]: is there a reason for using `LAMBDA-2.3.0` version as I see that we also already have `2.4.0` too; https://github.com/hearsttv/htv-terraform-modules/tree/LAMBDA-2.4.0/lambda-deploy

File: deploy/terraform/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "VAULT_USERNAME"  = data.vault_generic_secret.auto-clipping.data["username"]
+    "VAULT_PASSWORD"  = data.vault_generic_secret.auto-clipping.data["password"]
+  }
+}
+
+
+resource "aws_lambda_permission" "allow_result_bucket" {
+  statement_id  = "AllowExecutionFromS3Bucket"
+  action        = "lambda:InvokeFunction"
+  function_name = "${module.autoclipping_lambda.lambda_arn}"
```
Comment (filipe-cantarelli) [inline deploy/terraform/lambda.tf:23]: terraform12. Remove the brackets

File: deploy/terraform/lambda.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  principal     = "s3.amazonaws.com"
+  source_arn    = data.terraform_remote_state.enps.outputs.enps_bucket_arn
+}
+
+resource "aws_s3_bucket_notification" "s3_object_notification" {
+  bucket = data.terraform_remote_state.enps.outputs.enps_bucket_id
+
+  lambda_function {
+    lambda_function_arn = "${module.autoclipping_lambda.lambda_arn}"
```
Comment (filipe-cantarelli) [inline deploy/terraform/lambda.tf:32]: Terraform12

**Pull request merged**